backend/expense/src/main/java/org/egov/digit/expense/util/EnrichmentUtil.java

```diff
@@ -76,17 +78,20 @@ public void encrichBillForCreate(BillRequest billRequest) {
             billDetail.getPayee().setAuditDetails(audit);
             billDetail.getPayee().setStatus(Status.ACTIVE);
 
-            String gender = genderUtil.getGenderDetails(billRequest.getRequestInfo(),billDetail.getPayee().getTenantId(),billDetail.getPayee().getIdentifier());
-            ObjectMapper objectMapper = new ObjectMapper();
-            Map<String, Object> map = objectMapper.convertValue(billDetail.getPayee().getAdditionalDetails(), new TypeReference<Map<String, Object>>() {});
-            if(map == null){
-                map = new HashMap<>();
-            }
-            if(!gender.isEmpty()){
-                map.put(GENDER,gender);
+            if (!config.isHealthContextEnabled()) {
+                String gender = genderUtil.getGenderDetails(billRequest.getRequestInfo(), billDetail.getPayee().getTenantId(), billDetail.getPayee().getIdentifier());
+                ObjectMapper objectMapper = new ObjectMapper();
+                Map<String, Object> map = objectMapper.convertValue(billDetail.getPayee().getAdditionalDetails(), new TypeReference<Map<String, Object>>() {
+                });
+                if (map == null) {
+                    map = new HashMap<>();
+                }
+                if (!gender.isEmpty()) {
+                    map.put(GENDER, gender);
+                }
+                map.put(GENDER, gender);
```

Comment:
why is this repeating twice? you are adding if it's empty or not

Reply:
removed